public/js/neznam-atproto-share-comments.js

```diff
@@ -0,0 +1,135 @@
+  copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+  SOFTWARE.
+  */
+
+  const rootElement = document.querySelector('#neznam-atproto-share-comments')
+  if (!rootElement || !rootElement.dataset.uri) return
+  const atProto = rootElement.dataset.uri
+
+  fetch(
+    'https://public.api.bsky.app/xrpc/app.bsky.feed.getPostThread?uri=' + atProto
+  )
+    .then((response) => {
+      if (!response.ok) {
+        throw new Error(`HTTP error, status = ${response.status}`)
+      }
+      return response.json()
+    })
+    .then((data) => {
+      if (
+        typeof data.thread.replies !== 'undefined' &&
+        data.thread.replies.length > 0
+      ) {
+        rootElement.replaceChildren(renderComments(data.thread))
+        const someReplies = document.createElement('div')
+        someReplies.innerHTML = '<a href="' + ToBskyUrl(rootElement.dataset.uri) + '">Post a reply on BlueSky</a>'
+        rootElement.append(someReplies)
+      } else {
+        const noReplies = document.createElement('em')
+        noReplies.innerHTML = 'No replies. <a href="' + ToBskyUrl(rootElement.dataset.uri) + '">Post a reply on BlueSky</a>'
+        rootElement.replaceChildren(noReplies)
+      }
+    })
+    .catch((error) => {
+      console.warn(error)
+      const p = document.createElement('p')
+      p.appendChild(document.createTextNode(`Error: ${error.message}`))
+      document.body.appendChild(p, rootElement)
+    })
+
+  function ToBskyUrl (uri) {
+    const splitUri = uri.split('/')
+    if (splitUri[0] === 'at:') {
+      return 'https://bsky.app/profile/' + splitUri[2] + '/post/' + splitUri[4]
+    } else {
+      return uri
+    }
+  }
+
+  function renderComments (thread) {
+    const commentsNode = document.createElement('div')
+    for (const comment of thread.replies) {
+      const renderedString = renderComment(comment)
+      const htmlContent = createElementFromHTML(renderedString)
+
+      htmlContent.querySelector('.replies').appendChild(renderComments(comment))
+
+      commentsNode.appendChild(htmlContent)
+    }
+
+    return commentsNode
+  }
+
+  // https://stackoverflow.com/a/494348
+  function createElementFromHTML (htmlString) {
+    const div = document.createElement('div')
+    div.innerHTML = htmlString.trim()
+    return div.firstChild
+  }
+
+  function renderComment (comment) {
+    const replyDate = new Date(comment.post.record.createdAt)
+
+    return `<ul class="bluesky-comments">
+  <li class="avatar">
+    <img src="${comment.post.author.avatar}">

```

Comment:
Need to add check if user has avatar. While most have, if the user doesn't have we get undefined error.

Reply:
Good catch. I'm now checking for all attributes, even those that should always be present.